In [ ]:
Q1

In [23]:
%matplotlib inline
import numpy as np
import scipy as sp
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns
from statsmodels.stats.proportion import proportions_ztest

In [7]:
df=pd.read_csv("Cutlets.csv")
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [9]:
unitA=pd.Series(df.iloc[:,0])
unitA.head()

0    6.8090
1    6.4376
2    6.9157
3    7.3012
4    7.4488
Name: Unit A, dtype: float64

In [11]:
unitB=pd.Series(df.iloc[:,1])
unitB.head()

0    6.7703
1    7.5093
2    6.7300
3    6.7878
4    7.1522
Name: Unit B, dtype: float64

In [15]:
stats.ttest_ind(unitA,unitB)

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)

### pvalue=0.4722 is > alpha=0.05 (5% significance), then accept null hypothesis  ( Thus, there is no difference in diameters of cutlets between two units)

In [ ]:
Q2

In [17]:
df1=pd.read_csv("LabTAT.csv")
df1

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [19]:
Lab1=pd.Series(df1.iloc[:,0])
Lab2=pd.Series(df1.iloc[:,1])
Lab3=pd.Series(df1.iloc[:,2])
Lab4=pd.Series(df1.iloc[:,3])
Lab1,Lab2,Lab3,Lab4

(0      185.35
 1      170.49
 2      192.77
 3      177.33
 4      193.41
         ...  
 115    178.49
 116    176.08
 117    202.48
 118    182.40
 119    182.09
 Name: Laboratory 1, Length: 120, dtype: float64,
 0      165.53
 1      185.91
 2      194.92
 3      183.00
 4      169.57
         ...  
 115    170.66
 116    183.98
 117    174.54
 118    197.18
 119    215.17
 Name: Laboratory 2, Length: 120, dtype: float64,
 0      176.70
 1      198.45
 2      201.23
 3      199.61
 4      204.63
         ...  
 115    193.80
 116    215.25
 117    203.99
 118    194.52
 119    221.49
 Name: Laboratory 3, Length: 120, dtype: float64,
 0      166.13
 1      160.79
 2      185.18
 3      176.42
 4      152.60
         ...  
 115    172.68
 116    177.64
 117    170.27
 118    150.87
 119    162.21
 Name: Laboratory 4, Length: 120, dtype: float64)

In [24]:
 sp.stats.f_oneway(Lab1,Lab2,Lab3,Lab4)

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

### we accept null hypothesis Inference is that there no significant difference in the average TAT for all the labs.

In [ ]:
Q3

In [27]:
df2=pd.read_csv("BuyerRatio.csv")
df2

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [28]:
Male = [50,142,131,70]
Female=[435,1523,1356,750]
Sales=[Male,Female]
print(Sales)

[[50, 142, 131, 70], [435, 1523, 1356, 750]]


In [31]:
chiStats = sp.stats.chi2_contingency(Sales)
tstat=chiStats[0]
pvalue=chiStats[1]
tstat,pvalue

(1.595945538661058, 0.6603094907091882)

### Test t=1.595946 p-value=0.660309 Interpret by p-Value we accept null hypothesis

In [ ]:
Q4

In [33]:
df4=pd.read_csv('Costomer+OrderForm.csv')
df4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [35]:
Phillippines_value=df4['Phillippines'].value_counts()
Indonesia_value=df4['Indonesia'].value_counts()
Malta_value=df4['Malta'].value_counts()
India_value=df4['India'].value_counts()
print(Phillippines_value)
print(Indonesia_value)
print(Malta_value)
print(India_value)

Error Free    271
Defective      29
Name: Phillippines, dtype: int64
Error Free    267
Defective      33
Name: Indonesia, dtype: int64
Error Free    269
Defective      31
Name: Malta, dtype: int64
Error Free    280
Defective      20
Name: India, dtype: int64


In [39]:
chiStats2= sp.stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])
tstas=chiStats2[0]
pvalue=chiStats2[1]
tstas,pvalue

(3.858960685820355, 0.2771020991233135)

### Test t=3.858961 p-value=0.277102 Interpret by p-Value we accept null hypothesis